pkg/lib/packagekit.js

```diff
@@ -479,3 +520,219 @@ export function getBackendName() {
     return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
                 "Get", ["org.freedesktop.PackageKit", "BackendName"]);
 }
+
+/**
+ * Refresh PackageKit Cache
+ * @param {boolean} force - force refresh the cache (expensive)
+ * @param {*} progress_cb - progress callback
+ */
+export function refresh(force, progress_cb) {
+    force = force || false;
+    return cancellableTransaction("RefreshCache", [force], progress_cb);
+}
+
+/**
+ * Remove a package
+ *
+ * @param {?string[] | undefined} pkgnames - packages to remove
+ * @param {?string[] | undefined} files - files of a package to remove, used to resolve to a package
+ * @param {?() => void} progress_cb - progress callback
+ */
+export async function remove_packages(pkgnames, files, progress_cb) {
+    const promises = [];
+    const ids = [];
+
+    if (pkgnames) {
+        const flags = Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED | Enum.FILTER_NOT_SOURCE;
+        promises.push(cancellableTransaction("Resolve", [flags, pkgnames], progress_cb,
+                                             {
+                                                 Package: (_info, package_id) => ids.push(package_id),
+                                             }));
+    }
+
+    if (files) {
+        promises.push(cancellableTransaction("SearchFiles", [Enum.FILTER_INSTALLED, files], progress_cb,
+                                             {
+                                                 Package: (_info, package_id) => ids.push(package_id),
+                                             }));
+    }
+
+    await Promise.all(promises);
+
+    if (ids.length === 0)
+        return Promise.reject(new TransactionError("not-found", "Can't resolve packages(s)"));
+
+    return cancellableTransaction("RemovePackages", [0, ids, true, false], progress_cb);
+}
+
+/**
+ * @param {string[]} pkgnames - packages to install
+ * @param {?() => void} progress_cb - optional progress callback
+ */
+export async function install_packages(pkgnames, progress_cb) {
+    const flags = Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_NEWEST;
+    const to_installed_packages = new Set(pkgnames);
+    const ids = [];
+
+    // TODO: just always resolve all states?
+    await cancellableTransaction("Resolve", [flags | Enum.FILTER_INSTALLED, pkgnames], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => {
+                                         const pkg = package_id.split(";")[0];
+                                         if (to_installed_packages.has(pkg))
+                                             to_installed_packages.delete(pkg);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-52f3e529-20251111-125657-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)